lib/Semantic_check.ml

```diff
@@ -1,2163 +0,0 @@
-(** Semantic validation of AST*)
-
-(* Idea: check many of things related to identifiers that are hard to check
-during parsing and are in fact irrelevant for building up the parse tree *)
-
-(* To check:
-Variable in scope if used 
```

Comment:
Move these comments to the places where these actions occur

Reply:
Done.